app/services/participants/change_schedule/base.rb

```diff
@@ -1,16 +1,89 @@
 # frozen_string_literal: true
 
-require "abstract_interface"
-
 module Participants
   module ChangeSchedule
-    class Base < Participants::Base
-    private
+    class Base
+      include ProfileAttributes
+      include ActiveModel::Validations
+
+      attr_reader :schedule_identifier, :cohort
+
+      validates :schedule, presence: { message: I18n.t(:invalid_schedule) }
+      validate :not_already_withdrawn
+      validate :schedule_valid_with_pending_declarations
+      validate :validate_provider
 
       def initialize(params:)
-        super(params: params)
-        self.schedule_identifier = params[:schedule_identifier]
-        self.cohort = params[:cohort]
+        @participant_id = params[:participant_id]
+        @course_identifier = params[:course_identifier]
+        @cpd_lead_provider = params[:cpd_lead_provider]
+        @schedule_identifier = params[:schedule_identifier]
+        @cohort = params[:cohort]
+      end
+
+      def call
+        unless valid?
+          raise ActionController::ParameterMissing, errors.map(&:message)
+        end
+
+        ActiveRecord::Base.transaction do
+          ParticipantProfileSchedule.create!(participant_profile: user_profile, schedule: schedule)
+          user_profile.update_schedule!(schedule)
+        end
+
+        user_profile
+      end
+
+    private
+
+      def participant_profile_state
+        user_profile&.participant_profile_state
+      end
+
+      def validate_provider
+        unless matches_lead_provider?
+          errors.add(:participant_id, I18n.t(:invalid_participant))
+        end
+      end
+
+      def cohort_object
+        @cohort_object ||= if @cohort
+                             Cohort.find_by(start_year: @cohort)
+                           else
+                             Cohort.current
+                           end
+      end
+
+      def schedule
```

Comment:
Suggestion: the idiomatic way to do this and slightly, less contrived in my option.
```suggestion
        def schedule
          @schedule ||= begin
            alias_search_query = Finance::Schedule
              .where("identifier_alias IS NOT NULL")
              .where(identifier_alias: schedule_identifier, cohort: cohort_object)

            Finance::Schedule
              .where(schedule_identifier: schedule_identifier, cohort: cohort_object)
              .or(alias_search_query)
              .first
          end
        end
```

Reply:
sure we can change that next time around